core/Tracker/TrackerCodeGenerator.php

```diff
@@ -93,6 +115,7 @@ public function generate($idSite, $piwikUrl, $mergeSubdomains = false, $groupPag
 
         $codeImpl = array(
             'idSite'                  => $idSite,
+            // TODO why sanitizeInputValue() and not json_encode?
             'piwikUrl'                => Common::sanitizeInputValue($piwikUrl),
```

Comment:
The JS code is built w/ part of the whole URL (ie it's `'"' + $protocol + $url + '"'`), so json_encode wouldn't quite work. The sanitize call must be an explicit attempt at avoiding XSS, though could perhaps move the json_encode call below after concatenating the protocol and whatever else.